crypto/keyring/keyring.go

```diff
@@ -165,8 +165,7 @@ type Options struct {
 // NewInMemory creates a transient keyring useful for testing
 // purposes and on-the-fly key generation.
 // Keybase options can be applied when generating this new Keybase.
-func NewInMemory(opts ...Option) Keyring {
-	cdc := simappparams.EncodingConfig.Marshaler
+func NewInMemory(cdc codec.AminoCodec, opts ...Option) Keyring {
```

Comment:
We should not be using Amino anywhere. Here, it should be `codec.Codec`

Reply:
we need Amino for migrations - so only migration function is using Amino to deserialize old value.